libs/native/jna/src/main/java/org/elasticsearch/nativeaccess/jna/JnaPosixCLibrary.java

```diff
@@ -143,9 +144,17 @@ private interface FXStatFunction extends Library {
         this.functions = Native.load("c", NativeFunctions.class);
         FStat64Function fstat64;
         try {
+            // JNA lazily finds symbols, so even though we try to bind two different functions below, if fstat64
+            // isn't found, we won't know until runtime when calling the function. To force resolution of the
+            // symbol we get a function object directly from the native library. We don't use it, we just want to
+            // see if it will throw UnsatisfiedLinkError
+            NativeLibrary.getInstance("c").getFunction("fstat64");
             fstat64 = Native.load("c", FStat64Function.class);
         } catch (UnsatisfiedLinkError e) {
-            // TODO: explain
+            // fstat has a long history in linux from the 32-bit architecture days. On some modern linux systems,
```

Comment:
Does this make the class name `JnaPosixCLibrary` a bit of a misnomer?

Reply:
Maybe, but "fstat" is part of posix, it's just that the implementation is a bit wonky due to historical reasons (and thus why we need to sometimes peek into the implementation side).